tests/posts_tests/test_ratings.py

```diff
@@ -0,0 +1,44 @@
+import pytest
+from posts.models import Resume
+from pytest_django.asserts import assertTemplateUsed
+from conftest import RATER_USERNAME, RATER_PASSWORD, USERNAME, PASSWORD
+
+
+@pytest.mark.django_db()
+class TestRatingViews:
+    def test_get_rate_page_for_valid_resume(self, client, persist_resume, persist_rater):
+        resume = persist_resume
+        client.login(username=RATER_USERNAME, password=RATER_PASSWORD)
+        response = client.get(f'/post/{resume.pk}/rate/')
+        assert response.status_code == 200
+        assertTemplateUsed(response, 'posts/rating_form.html')
+
+    @pytest.mark.parametrize('invalid_resume', ['-1', '', 'resume'])
+    def test_get_rate_page_for_invalid_resume_path(self, client, invalid_resume, persist_rater):
+        client.login(username=RATER_USERNAME, password=RATER_PASSWORD)
+        response = client.get(f'/post/{invalid_resume}/rate/')
+        assert response.status_code == 404
+
+    def test_get_rate_page_for_not_exist_resume(self, client):
+        last_resume = Resume.objects.latest('post_id')
+        client.login(username=RATER_USERNAME, password=RATER_PASSWORD)
+        with pytest.raises(Resume.DoesNotExist):
+            client.get(f'/post/{last_resume.post_id+1}/rate/')
```

Comment:
Is this an error that the user should see?

Reply:
@natifridman 

Because the tests are using `Client` to test responses from the server, it throws this error when trying access a view page related an object that is not exist.
(I cat the console)
![image](https://user-images.githubusercontent.com/36789773/168596098-f95db1c1-356b-4bf9-9966-23812a443ec8.png)

When a user tries to access a post that is not exist(invalid ID) it will show him status code 404 with the exception.

![image](https://user-images.githubusercontent.com/36789773/168595970-b1d54d13-ee2a-4710-a8cc-709dc8db842c.png)

Reply:
For our use case that's ok. Keep in mind that we usually don't want to show this error to our users for various reasons, one of them is security.